### PR TITLE
Mono cmake build work.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -856,6 +856,8 @@
       <Output TaskParameter="ExitCode" PropertyName="MonoNinjaFound"/>
     </Exec>
     <PropertyGroup>
+      <_MonoVerboseArg Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(MonoNinjaFound)' == '0'">-v</_MonoVerboseArg>
+      <_MonoVerboseArg Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(MonoNinjaFound)' != '0'">VERBOSE=1</_MonoVerboseArg>
```
Comment (akoeplinger) [inline src/mono/mono.proj:860]: I don't think we should turn on verbose build in CI by default, it just bloats the build log and makes it harder to understand what's going on.

I'm fine with having the option of course so you can enable it when something goes wrong :)

File: src/mono/mono.proj [hunk 2]
```diff
@@ -881,6 +883,8 @@
 	  <_MonoCMakeArgs Include="-DCMAKE_TOOLCHAIN_FILE=$(RepositoryEngineeringDir)/common/cross/toolchain.cmake" />
       <_MonoBuildEnv Condition="'$(Platform)' == 'arm64'" Include="TARGET_BUILD_ARCH=arm64" />
       <_MonoBuildEnv Condition="'$(Platform)' == 'arm'" Include="TARGET_BUILD_ARCH=arm" />
+      <_MonoBuildEnv Condition="'$(Platform)' == 'arm64'" Include="PKG_CONFIG_PATH=$(ROOTFS_DIR)/usr/lib/aarch64-linux-gnu/pkgconfig" />
+      <_MonoBuildEnv Condition="'$(Platform)' == 'arm'" Include="PKG_CONFIG_PATH=$(ROOTFS_DIR)/usr/lib/arm-linux-gnueabihf/pkgconfig" />
```
Comment (akoeplinger) [inline src/mono/mono.proj:887]: should be:
```suggestion
      <_MonoBuildEnv Condition="'$(Platform)' == 'arm64'" Include="PKG_CONFIG_PATH=$(MonoCrossDir)/usr/lib/aarch64-linux-gnu/pkgconfig" />
      <_MonoBuildEnv Condition="'$(Platform)' == 'arm'" Include="PKG_CONFIG_PATH=$(MonoCrossDir)/usr/lib/arm-linux-gnueabihf/pkgconfig" />
```

**Comment by vargaz** [comment]
Updated.

**Comment by akoeplinger** [comment]
Replaced by https://github.com/dotnet/runtime/pull/42119

**Pull request closed without merging**